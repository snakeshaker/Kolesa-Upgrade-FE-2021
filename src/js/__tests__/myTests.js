describe('Проверка', () => {
    const appElement = document.createElement('div');
    const loader = document.createElement('div');
    const btn = document.createElement('button');
    const eventHandler = jest.fn();

    appElement.setAttribute('id', 'app');
    loader.setAttribute('id', 'loader');
    const insert = elem => document.body.appendChild(elem);
    const remove = elem => elem.remove();

    beforeAll(() => {
        insert(appElement);
        insert(loader);
        insert(btn);
        btn.addEventListener('click', eventHandler);
        btn.click();
    });

    afterAll(() => {
        remove(appElement);
    });

    it('Тест на добавление контента на страницу в блок #app', () => {
        appElement.innerHTML += 'Контент добавлен!';
        expect(appElement).toMatchSnapshot();
    });

    it('Тест на отображение и скрытие лоадера', () => {
        expect(document.getElementById('loader')).toBeTruthy();
        remove(loader);
        expect(document.getElementById('loader')).toBeFalsy();
    });

    it('Тест на добавление событий', () => {
        expect(eventHandler).toHaveBeenCalled();
    });
});
