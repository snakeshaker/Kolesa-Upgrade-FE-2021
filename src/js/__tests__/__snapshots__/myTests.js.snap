// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка Тест на добавление контента на страницу в блок #app 1`] = `
<div
  id="app"
>
  Контент добавлен!
</div>
`;
